### PR TITLE
Typescript improvements

### PR DESCRIPTION
**Pull request by jescalan**
I switched files over to typescript previously but didn't properly instrument the types - this PR adds types out of the box. Also took the opportunity to upgrade to the new broken up nextjs-scripts packages.

I am getting an issue with a postcss plugin though which is unusual - I got it with the previous and current version of nextjs-scripts so it does not have to do with the broken up packages swap. I just tested and the error is present in the current main branch too, which is weird.

### CODE REVIEW

File: components/button/index.tsx [hunk 1]
```diff
@@ -3,7 +3,15 @@
...
-export default function Button({ children, href }) {
+export interface ButtonProps {
+  href: string
+  children: React.ReactChildren
+}
+
+export default function Button({
+  children,
+  href,
+}: ButtonProps): React.ReactElement {
```
Comment (BRKalow) [inline components/button/index.tsx:14]: I think ideally if we're typing a component that can accept `children` we look to leverage the `React.FC` helper type. We'd have to change the definition to `const`:

```tsx
interface ButtonProps {
  href: string
}

const Button: React.FC<ButtonProps> = ({ children, href }) => { ... }
```

File: pages/people/index.tsx [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
-export default function PeoplePage({ allPeople, allDepartments }) {
+interface Props {
+  allPeople: PersonRecord[]
+  allDepartments: DepartmentRecord[]
+}
+
+export default function PeoplePage({
+  allPeople,
+  allDepartments,
+}: Props): React.ReactElement {
```
Comment (BRKalow) [inline pages/people/index.tsx:17]: Kinda nit, but typing the return type of a component like this seems unnecessary. We never consume a page and care about its return value directly.

File: components/button/index.tsx (hunk 1 above)
Comment (jescalan) [reply components/button/index.tsx:14]: What is the advantage of doing this?

File: pages/people/index.tsx (hunk 2 above)
Comment (jescalan) [reply pages/people/index.tsx:17]: Agreed - I'd like to change this in our default eslint config though if that's where we want to go 😀

**Comment by jescalan** [comment]
Merging to fix bugz but happy to follow up on any of these comments still!

**Pull request merged**